pandas/core/arrays/datetimes.py

```diff
@@ -207,26 +215,46 @@ def __new__(cls, values, freq=None, tz=None, dtype=None):
         # if dtype has an embedded tz, capture it
         tz = dtl.validate_tz_from_dtype(dtype, tz)
 
-        if isinstance(values, DatetimeArrayMixin):
+        if lib.is_scalar(values):
+            raise ValueError('{cls}() must be called with a '
+                             'collection of some kind, {data} was passed'
+                             .format(cls=cls.__name__, data=repr(values)))
+        elif isinstance(values, ABCSeries):
+            # extract nanosecond unix timestamps
+            if tz is None:
+                # TODO: Try to do this in just one place
+                tz = values.dt.tz
+            values = np.array(values.view('i8'))
+        elif isinstance(values, DatetimeArrayMixin):
             # extract nanosecond unix timestamps
             values = values.asi8
+
         if values.dtype == 'i8':
             values = values.view('M8[ns]')
 
         assert isinstance(values, np.ndarray), type(values)
         assert is_datetime64_dtype(values)  # not yet assured nanosecond
-        values = conversion.ensure_datetime64ns(values, copy=False)
+        values = conversion.ensure_datetime64ns(values, copy=copy)
 
         result = cls._simple_new(values, freq=freq, tz=tz)
-        if freq_infer:
-            inferred = result.inferred_freq
-            if inferred:
-                result.freq = to_offset(inferred)
+        dtl.maybe_define_freq(freq_infer, result)
 
         # NB: Among other things not yet ported from the DatetimeIndex
         # constructor, this does not call _deepcopy_if_needed
         return result
 
+    @classmethod
+    def _from_sequence(cls, scalars, dtype=None, copy=False):
+        # list, tuple, or object-dtype ndarray/Index
```

Comment:
You're right we could make do without it.  I like doing this explicitly because to_datetime is already overloaded and circular.

Reply:
this is horribly inefficient and unnecessary

Reply:
If we don't do it here, to_datetime is going to do this.  It may be unnecessary, but it is not horribly inefficient.  What _is_ a code smell is the circularity involved in calling to_datetime.

Reply:
then just call array_to_datetime and don’t force the conversion to array

Reply:
array_to_datetime expects an object ndarray.  I'll see about calling array_to_datetime directly instead of to_datetime.

Reply:
> So is the root problem (referenced in your "circularity" comment, and down below in TimedeltaIndex.__new__) that to_datetime / to_timedelta returns an Index instead of an EA?

It's not the fact that it's an Index so much as that it is a circular dependency.  I think I can resolve this in an upcoming commit.

> Looking through to_datetime and _convert_listlike_datetimes, I don't see a conversion to ndarray[object].

_convert_listlike_datetimes calls `ensure_object`.

> Sorry, to_datetime has in intermediate datetime64[ns] -> object -> datetime64[ns] conversion? That seems unnecessary.

Not sure what you're referring to.  As implemented _from_sequence is specifically for list, tuple, or object-dtype NDArray/Index.  datetime64-dtype goes through a different path.

Reply:
> _convert_listlike_datetimes calls ensure_object.

That's after an

```python
    # these are shortcutable
    if is_datetime64tz_dtype(arg):
        if not isinstance(arg, DatetimeIndex):
            return DatetimeIndex(arg, tz=tz, name=name)
        if tz == 'utc':
            arg = arg.tz_convert(None).tz_localize(tz)
        return arg

    elif is_datetime64_ns_dtype(arg):
        if box and not isinstance(arg, DatetimeIndex):
            try:
                return DatetimeIndex(arg, tz=tz, name=name)
            except ValueError:
                pass

        return arg
```

So those both avoid conversion to object.

Reply:
`ExtensionArray._from_sequence` is for any sequence of scalar objects, including a ndarray with a specialized type like `datetime64[ns]`. It'll be used, for example, in `factorize(ExtensionArray)`.